src/foam/dao/mongodb/MongoDBService.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * @license
+ * Copyright 2025 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.dao.mongodb',
+  name: 'MongoDBService',
+
+  documentation: 'Context service for MongoDAOs to acquire connection specs.',
+
+  // Future: MongoDB supports change notification.  A listener can be
+  // notified of database or collection chaanges.  This can be used
+  // in a cluster setting to update the MDAOs when an change is
+  // reported.  
+  // see: https://www.mongodb.com/docs/drivers/java/sync/current/usage-examples/watch/
+  // When the MongoDAOs startup they could register themselves with this
+  // service which could in turn issue the appropriate find and put
+  // operations with the updated data. 
+  
+  implements: [
+    'foam.core.COREService'
+  ],
+
+  javaImports: [
+    'foam.core.logger.Logger',
+    'foam.core.logger.Loggers',
+    'foam.lang.X',
+    'com.mongodb.*',
+    'com.mongodb.client.MongoDatabase',
+    'com.mongodb.client.MongoClient',
+    'com.mongodb.client.MongoClients',
+    'org.bson.BsonDocument',
+    'org.bson.BsonInt64',
+    'org.bson.Document',
+    'org.bson.conversions.Bson',
+  ],
+
+  javaCode: `
+    private MongoClient client_;
+    private void setClient(MongoClient client) { client_ = client; }
+    private MongoClient getClient() { return client_; }
+  `,
+
+  properties: [
+    {
+      name: 'dbName',
+      class: 'String',
+      value: 'foam'
+    },
+    {
+      name: 'url',
+      class: 'String',
+      value: 'mongodb://localhost:27017/'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'getDatabase',
+      args: 'X x, String collectionName',
+      javaType: 'MongoDatabase',
+      javaCode: `
+      return getClient().getDatabase(getDbName());
+      `
+    },
+    {
+      name: 'start',
+      javaCode: `
+        initialize(getX());
+      `
+    },
+    {
+      name: 'reload',
+      javaCode: `
+        MongoClient client = getClient();
+        initialize(getX());
+        if ( client != null ) {
+          client.close();
+        }
+      `,
+    },
+    {
+      name: 'stop',
+      javaCode: `
+        MongoClient client = getClient();
+        if ( client != null ) {
+          client.close();
+        }
```

Comment:
MongoDBService keeps an open client connection to mongodb. Can we call `stop()` on graceful shutdown to close the connection?

Just a thought, I'm thinking of adding a shutdown hook in Boot.java to call `stop()` on instances of COREService.

Reply:
yes, shutdown hook has been on my todo for some time, and it was the reason I added 'stop' to COREService.  Will add.

Reply:
@lchanmann see shutdown hook: https://github.com/kgrgreer/foam3/pull/3740